core/src/main/java/com/radixdlt/monitoring/MetricInstaller.java

```diff
@@ -66,30 +66,72 @@
 
 import com.google.inject.Inject;
 import com.radixdlt.consensus.bft.BFTNode;
+import com.radixdlt.consensus.bft.BFTValidatorSet;
 import com.radixdlt.consensus.bft.Self;
 import com.radixdlt.monitoring.Metrics.Config;
+import com.radixdlt.p2p.PeersView;
+import java.util.Collection;
 
 /** An installer of extra metrics which do not follow the conventional Prometheus usage patterns. */
 public final class MetricInstaller {
 
   /** An own node, for exposing the {@link Config#key()} information. */
   private final BFTNode self;
 
+  /** A source of "system" getters to be exposed as gauges. */
+  private final InMemorySystemInfo inMemorySystemInfo;
+
+  /** A source of "peers" getters to be exposed as gauges. */
+  private final PeersView peersView;
+
   @Inject
-  public MetricInstaller(@Self BFTNode self) {
+  public MetricInstaller(
+      final @Self BFTNode self,
+      final InMemorySystemInfo inMemorySystemInfo,
+      final PeersView peersView) {
     this.self = self;
+    this.inMemorySystemInfo = inMemorySystemInfo;
+    this.peersView = peersView;
   }
 
   /**
    * Sets up the metrics which - for different reasons (most often legacy) - do not use the regular
    * Prometheus measurement primitives.
    *
-   * <p>This includes e.g. static "info" metrics, and directly-read "reader gauges".
+   * <p>This includes e.g. static "info" metrics, and directly-read "getter gauges".
    *
    * @param metrics Hierarchy where some legacy metrics need to be set.
    */
   public void installAt(Metrics metrics) {
-    var config = new Config(ApplicationVersion.INSTANCE.string(), this.self.getKey().toHex());
+    final var config = new Config(ApplicationVersion.INSTANCE.string(), this.self.getKey().toHex());
     metrics.misc().config().set(config);
+    metrics.peers().peerCount().initialize(() -> this.peersView.peers().count());
```

Comment:
not a big deal but we could save some computation by adding a dedicated `count` method in PeersView; its main impl (`PeerManagerPeersView`) does some extra grouping and list allocation that could be skipped

Reply:
I bet there are a few similar instances (where an unnecessarily complex computation is performed to gather metrics), but to be on the safe side, I prefer to first "simply" migrate the existing metric as-is, without trying to optimize.

_(after all, even if this is veeeery sub-optimal, it will be executed only once per 30s-5min [depending on Prometheus scraping config])_